benchmarks/benchmark.py

```diff
@@ -97,14 +107,52 @@ def benchmark_suite(prepare: Callable[..., Callable], params_list: List[Dict],
     times.append(benchmark(f, name=subname,
                            target_total_secs=target_total_secs))
 
-  print("---------Benchmark summary for %s---------" % name)
   param_names = list(params_list[0].keys())
-  print(tabulate([tuple(map(_param_str, params.values())) +
-                  (t.mean(), _pstd(t), t.mean() / times[0].mean())
-                  for params, t in safe_zip(params_list, times)],
-                 param_names + ["mean", "%std", "relative"]))
+  data_header = param_names + ["mean", "%std", "relative"]
+  data = [list(map(_param_str, params.values())) +
+          [t.mean(), _pstd(t), t.mean() / times[0].mean()]
+          for params, t in safe_zip(params_list, times)]
+
+  if FLAGS.baseline_dir:
+    mean_idx = len(param_names)
+    means = _get_baseline_means(mean_idx, FLAGS.baseline_dir, name)
+    assert len(means) == len(data), (means, data)
+    data_header.append("mean/baseline")
+    for idx, mean in enumerate(means):
+      data[idx].append(data[idx][mean_idx] / mean)
+
+  print("---------Benchmark summary for %s---------" % name)
+  print(tabulate(data, data_header))
   print()
 
+  if FLAGS.export_dir:
+    filename = _export_results(data_header, data, FLAGS.export_dir, name)
+    print("Wrote %s results to %s" % (name, filename))
+    print()
+
+
+def _get_baseline_means(mean_idx, baseline_dir, name):
+  baseline_dir = os.path.expanduser(baseline_dir)
+  filename = os.path.join(baseline_dir, name + ".csv")
+  if not os.path.exists(filename):
+    raise FileNotFoundError("Can't find baseline file: %s" % filename)
+  with open(filename, newline="") as csvfile:
+    reader = csv.reader(csvfile)
+    header = next(reader)
+    assert header[mean_idx] == "mean", (header, mean_idx)
```

Comment:
Wouldn't this code be more resilient if instead of taking a `mean_idx` parameter it would require one header element to be  the word "mean"?  And then I would put an assert in the `_export_results`.

Reply:
Good idea, done!